### ISSUE TITLE
Dependency error with ~2.0

### ISSUE THREAD

**Issue by alexislefebvre** [Troopers/AlertifyBundle#66]
Victoire 2.2 [requires the `~2.0` version](https://github.com/Victoire/victoire/blob/f19f0d1601be3493586d665391d5842cf103b8c3/composer.json#L47).

I can't install this version when simulating a fresh Symfony 2.8 installation :

```bash
$ composer require symfony/symfony:"2.8.*" --no-update
./composer.json has been created
$ composer require troopers/alertify-bundle:~2.0
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - troopers/alertify-bundle 2.0.3 requires friendsofsymfony/jsrouting-bundle 2.0.*@dev -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - troopers/alertify-bundle 2.0.2 requires friendsofsymfony/jsrouting-bundle 2.0.*@dev -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - troopers/alertify-bundle 2.0.1 requires friendsofsymfony/jsrouting-bundle 2.0.*@dev -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - troopers/alertify-bundle 2.0.0 requires friendsofsymfony/jsrouting-bundle 2.0.*@dev -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - Installation request for troopers/alertify-bundle ~2.0 -> satisfiable by troopers/alertify-bundle[2.0.0, 2.0.1, 2.0.2, 2.0.3].


Installation failed, reverting ./composer.json to its original content.
```

It looks like AlertifyBundle (`2.0.3`) is [requiring a `dev` version of FOSJsRoutingBundle](https://github.com/Troopers/AlertifyBundle/blob/04e1d02e5cdb68560935ad8f555e18b801684985/composer.json#L21) and that it break dependencies. 

FOSJsRoutingBundle does not provide any stable release: https://packagist.org/packages/friendsofsymfony/jsrouting-bundle

Maybe we can try to change the required version, as suggested here: https://github.com/FriendsOfSymfony/FOSJsRoutingBundle/issues/240#issuecomment-197275013

**Comment by alexislefebvre** [comment]
As noted in the [Composer documentation](https://getcomposer.org/doc/04-schema.md#package-links):

> If one of your dependencies has a dependency on an unstable package you need to explicitly require it as well, along with its sufficient stability flag.

And it works:

```bash
$ composer require symfony/symfony:"2.8.*" --no-update
./composer.json has been created
$ composer require friendsofsymfony/jsrouting-bundle:"~2.0@dev" --no-update
./composer.json has been updated
$ composer require troopers/alertify-bundle:~2.0
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 23 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-mbstring (v1.3.0) Loading from cache
  - Installing twig/twig (v2.0.0) Loading from cache
[…]
  - Installing willdurand/jsonp-callback-validator (v1.1.0) Loading from cache
  - Installing friendsofsymfony/jsrouting-bundle (dev-master 31462ad) Cloning 31462ad469 from cache
  - Installing troopers/alertify-bundle (2.0.3) Downloading: 100%         
symfony/security-acl suggests installing doctrine/dbal (For using the built-in ACL implementation)
paragonie/random_compat suggests installing ext-libsodium (Provides a modern crypto API that can be used to generate random bytes.)
troopers/alertify-bundle suggests installing troopers/assetic-injector-bundle (~1.1)
Writing lock file
Generating autoload files
```

**Comment by lenybernard** [comment]
This is fixed since 50fa32b11507177b73b253b0859fbc76eefd9b9e
